### ISSUE TITLE
Missing `zlib.so.1` error after commit `1fd2dc01ee34817858261bebae6a79cb9debc1f9`

### ISSUE THREAD

**Issue by luizirber** [oxalica/rust-overlay#63]
First, thanks for the great project, I've been using it for months without any issue =]

That said, I've been having trouble updating after commit 1fd2dc01ee34817858261bebae6a79cb9debc1f9 because `zlib.so.1` can't be found.

Steps to reproduce the issue:

```
$ git clone https://github.com/luizirber/decoct 
$ git checkout 405cdb341b8dc8cfb1ad1a44e3205c2efbff6934
$ nix-shell  # works
$ niv update rust-overlay -r 1fd2dc01ee34817858261bebae6a79cb9debc1f9
$ nix-shell  # fails, missing zlib.so.1
```

<details>
   <summary>Full log:</summary>

    these derivations will be built:                                                                                                                                                      
      /nix/store/clrf4gcpc3aq3d4ypyivcyd84a02bnmp-rust-1.56.1-x86_64-unknown-linux-gnu.drv
      /nix/store/rsvxlmrqsgr4m1zxd4zsyhw4d87hd36d-rust-legacy-1.56.1.drv
    building '/nix/store/clrf4gcpc3aq3d4ypyivcyd84a02bnmp-rust-1.56.1-x86_64-unknown-linux-gnu.drv'...
    unpacking sources
    unpacking source archive /nix/store/np4qhsdd77fr68v9g1ym6bmlch38d4ji-rust-1.56.1-x86_64-unknown-linux-gnu.tar.xz
    source root is rust-1.56.1-x86_64-unknown-linux-gnu
    setting SOURCE_DATE_EPOCH to timestamp 1635755669 of file rust-1.56.1-x86_64-unknown-linux-gnu/version
    patching sources
    installing
    Installing component rustc
    Installing component rust-std-x86_64-unknown-linux-gnu
    Installing component rust-docs
    Installing component rust-demangler-preview
    Installing component cargo
    Installing component rustfmt-preview
    Installing component rls-preview
    Installing component llvm-tools-preview
    Installing component clippy-preview
    Installing component rust-analysis-x86_64-unknown-linux-gnu
    post-installation fixup
    shrinking RPATHs of ELF executables and libraries in /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/libexec/cargo-credential-1password
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.56.1-stable.so
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libtest-7ddc9ce4941507e1.so
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/rust-lld
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-dis
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-cov
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-as
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-nm
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-size
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/gcc-ld/ld
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/gcc-ld/ld64
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llc
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-strip
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/opt
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-objcopy
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/rust-llvm-dwp
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-profdata
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-readobj
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-objdump
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-ar
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-65a28bf1738424c0.so
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.56.1-stable.so
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libtest-7ddc9ce4941507e1.so
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/librustc_driver-1a7f7f6d0c5586ea.so
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libstd-65a28bf1738424c0.so
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/bin/cargo-clippy
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/bin/rustdoc                                                                                 [70/344]
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/bin/rustc
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/bin/clippy-driver
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/bin/rust-demangler
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/bin/rls
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/bin/cargo-fmt
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/bin/rustfmt
    shrinking /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/bin/cargo
    gzipping man pages under /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/share/man/
    patching script interpreter paths in /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu
    /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/bin/rust-lldb: interpreter directive changed from "#!/bin/sh" to "/nix/store/l0wlqpbsvh1pgvhcdhw7qkka3d31si7k-
    bash-5.1-p8/bin/sh"
    /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/bin/rust-gdbgui: interpreter directive changed from "#!/bin/sh" to "/nix/store/l0wlqpbsvh1pgvhcdhw7qkka3d31si7
    k-bash-5.1-p8/bin/sh"
    /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/bin/rust-gdb: interpreter directive changed from "#!/bin/sh" to "/nix/store/l0wlqpbsvh1pgvhcdhw7qkka3d31si7k-b
    ash-5.1-p8/bin/sh"
    checking for references to /build/ in /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu...
    automatically fixing dependencies for ELF files
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/libexec/cargo-credential-1password
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.56.1-stable.so
      libz.so.1 -> not found!
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libtest-7ddc9ce4941507e1.so
      libstd-65a28bf1738424c0.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-65a28bf1738424c0.so
    setting RPATH to: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/rust-lld
      libLLVM-13-rust-1.56.1-stable.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.56.1-stable.so
    setting RPATH to: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-dis
      libLLVM-13-rust-1.56.1-stable.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.56.1-stable.so
    setting RPATH to: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-cov
      libLLVM-13-rust-1.56.1-stable.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.56.1-stable.so
    setting RPATH to: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-as
      libLLVM-13-rust-1.56.1-stable.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.56.1-stable.so
    setting RPATH to: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-nm
      libLLVM-13-rust-1.56.1-stable.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.56.1-stable.so
    setting RPATH to: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-size
      libLLVM-13-rust-1.56.1-stable.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.56.1-stable.so
    setting RPATH to: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/gcc-ld/ld
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/gcc-ld/ld64
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llc
      libLLVM-13-rust-1.56.1-stable.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.56.1-stable.so
    setting RPATH to: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-strip
      libLLVM-13-rust-1.56.1-stable.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.56.1-stable.so
    setting RPATH to: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/opt
      libLLVM-13-rust-1.56.1-stable.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.56.1-stable.so
    setting RPATH to: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-objcopy
      libLLVM-13-rust-1.56.1-stable.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.56.1-stable.so
    setting RPATH to: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/rust-llvm-dwp
      libLLVM-13-rust-1.56.1-stable.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.56.1-stable.so
    setting RPATH to: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-profdata
      libLLVM-13-rust-1.56.1-stable.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.56.1-stable.so
    setting RPATH to: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-readobj
      libLLVM-13-rust-1.56.1-stable.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.56.1-stable.so
    setting RPATH to: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-objdump
      libLLVM-13-rust-1.56.1-stable.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.56.1-stable.so
    setting RPATH to: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-ar
      libLLVM-13-rust-1.56.1-stable.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.56.1-stable.so
    setting RPATH to: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-65a28bf1738424c0.so
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.56.1-stable.so
      libz.so.1 -> not found!
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libtest-7ddc9ce4941507e1.so
      libstd-65a28bf1738424c0.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-65a28bf1738424c0.so
    setting RPATH to: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/librustc_driver-1a7f7f6d0c5586ea.so
      libstd-65a28bf1738424c0.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-65a28bf1738424c0.so
      libLLVM-13-rust-1.56.1-stable.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.56.1-stable.so
    setting RPATH to: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib:/nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.
    56.1-x86_64-unknown-linux-gnu/lib
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libstd-65a28bf1738424c0.so
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/bin/cargo-clippy
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/bin/rustdoc
      libtest-7ddc9ce4941507e1.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/libtest-7ddc9ce4941507e1.so
      librustc_driver-1a7f7f6d0c5586ea.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/librustc_driver-1a7f7f6d0c5586ea.so
      libstd-65a28bf1738424c0.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-65a28bf1738424c0.so
    setting RPATH to: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib:/nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib:/ni
    x/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/bin/rustc
      librustc_driver-1a7f7f6d0c5586ea.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/librustc_driver-1a7f7f6d0c5586ea.so
      libstd-65a28bf1738424c0.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-65a28bf1738424c0.so
    setting RPATH to: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib:/nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rus
    tlib/x86_64-unknown-linux-gnu/lib
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/bin/clippy-driver
      librustc_driver-1a7f7f6d0c5586ea.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/librustc_driver-1a7f7f6d0c5586ea.so
      libstd-65a28bf1738424c0.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-65a28bf1738424c0.so
    setting RPATH to: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib:/nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rus
    tlib/x86_64-unknown-linux-gnu/lib
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/bin/rust-demangler
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/bin/rls
      librustc_driver-1a7f7f6d0c5586ea.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/librustc_driver-1a7f7f6d0c5586ea.so
      libstd-65a28bf1738424c0.so -> found: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-65a28bf1738424c0.so
    setting RPATH to: /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib:/nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rus
    tlib/x86_64-unknown-linux-gnu/lib
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/bin/cargo-fmt
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/bin/rustfmt
    searching for dependencies of /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/bin/cargo
    autoPatchelfHook could not satisfy dependency libz.so.1 wanted by /nix/store/4yhrxcfpy20lbsnk399s21q75nyn8b63-rust-1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/lib
    LLVM-13-rust-1.56.1-stable.so
    Add the missing dependencies to the build inputs or set autoPatchelfIgnoreMissingDeps=true
    builder for '/nix/store/clrf4gcpc3aq3d4ypyivcyd84a02bnmp-rust-1.56.1-x86_64-unknown-linux-gnu.drv' failed with exit code 1
    cannot build derivation '/nix/store/rsvxlmrqsgr4m1zxd4zsyhw4d87hd36d-rust-legacy-1.56.1.drv': 1 dependencies couldn't be built
    error: build of '/nix/store/rsvxlmrqsgr4m1zxd4zsyhw4d87hd36d-rust-legacy-1.56.1.drv' failed
</details>

**Comment by VergeDX** [comment]
#62

**Comment by VergeDX** [comment]
Fixed in #62.

**Comment by oxalica** [comment]
Fixed in #62. Also worth mentioning that from your log, you are using the legacy component `rust-bin.<channel>.<version>.rust`, while it's encouraged to use `rust-bin.<channel>.version.default` or `minimal`.
